### ISSUE TITLE
Rust quality: three MLAS test steps report ok on a filter that matches nothing

### ISSUE THREAD

**Issue by justinchuby** [justinchuby/onnx-genai#2055]
`#2048` closed this hole for the Miri lane. The same hole is open in **`Rust quality`**, on the three steps that are the *only* place real-MLAS coverage runs.

## The exposure, audited exhaustively

All 17 non-Miri `cargo test` invocations across `.github/workflows/` fall into two groups:

**Loud already — `--test <target>` (12 steps).** A stale target name is a hard error, verified:

```
$ cargo test --locked -p onnx-runtime-ep-cpu --test no_such_target_xyz
error: no test target named `no_such_target_xyz`.
Available test targets: ... task_runtime_latency, tensor_scatter_reference_parity
rc=101
```

**Silent — a bare name/module filter (3 steps), all in `Rust quality`:**

| step | filter | tests it selects today |
|---|---|---|
| `ci.yml:425` Check CPU EP MLAS-only feature configuration | `optimization_registry_excludes_nchwc_without_cnn_ops` | **1** |
| `ci.yml:439` CPU EP MoE tests against real MLAS | `kernels::moe::` | **19** |
| `ci.yml:452` CPU EP QLinearMatMul tests against real MLAS | `kernels::qlinear_matmul::` | **30** |

(Arities measured on `main` `55b765cf8`, not read off the source.)

## Demonstrated, not inferred

```
$ cargo test --locked -p onnx-runtime-ep-cpu --no-default-features --features mlas kernels::moe_renamed::
test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 1731 filtered out; finished in 0.00s
rc=0
```

One rename away: green tick, `ok`, zero tests, and `set -euo pipefail` cannot see it because nothing failed — the step genuinely succeeded at running nothing.

## Why these three specifically matter

Their own comments say it. `kernels::moe::` exists because *"under the default feature set those tests compile the non-MLAS fallback, so the path that actually ships was going untested"* — the grouped MoE driver hands several `*mut f32` out of one `&mut [f32]` to `MlasGemmBatch`, and those bit-exactness falsifiers are the only thing between a wrong batch offset and silently wrong logits. `kernels::qlinear_matmul::` is the same argument: *"the coverage lane builds this crate without MLAS and the ORT lane tests the plugin crate, not these unit tests."*

So the failure isn't "a step stops running". It is **real-MLAS numerical coverage disappearing entirely, with nothing anywhere reporting its absence**, on exactly the paths that ship. The step's comment would still be sitting there explaining why the coverage is essential, next to a green tick, describing tests that no longer run.

`ci.yml:425` is the most fragile of the three — an exact single-test name, so *any* edit to that test's name silences it.

## Suggested fix

The logic already exists and is already conformance-tested; only its packaging is Miri-specific. Rename `scripts/miri_step.sh` → `scripts/test_step.sh` exposing a generic

```
run_test_step "<label>" <command...>
```

and keep `run_miri` as a two-line shim over it so `miri.yml` and its `MIRI_EXECUTED` / `MIRI_TIMING` output are unchanged. Then wrap the three steps above. `scripts/miri_step_test.sh` extends with a cell per new call site; it is hermetic (no cargo, no Miri) and runs in well under a second, so the cost to `Rust quality` is unchanged.

Definition of "executed" should stay exactly as `#2048` set it: `passed + failed >= 1` summed across every `test result:` line, **ignored does not count**, and **no opt-out** — an opt-out is the hole.

## Why an issue and not a PR right now

The patch is small and I have the audit done, but the runner pool is at **50 queued / 14 in progress** and `main` has 14 commits with no completed CI. This is hardening, not a regression — all three filters match today. Adding ~12 jobs to a queue that is starving real fixes is the wrong trade tonight. Happy to open it the moment the queue drains, or for anyone else to take it.

/cc @justinchuby — filed off the back of #2048.

**Comment by justinchuby** [comment]
PR up: #2066 (ready, normal auto-merge armed, no bypass — currently `BLOCKED` on required checks, as it should be).

Two things I learned building it that change what I filed here.

**The fix I proposed in this issue was too narrow.** Wrapping the three known steps fixes #2055 for exactly three steps and reopens the moment a fourth filtered step is added — which is how the hole got here. #2066 adds a scanner over `ci.yml` that fails on *any* `cargo test` step carrying a bare positional filter outside the guard. The subject is right: `-p` and `--test` both exit 101 on no match, so an option-only step needs no guard; the bare positional is the only selector libtest doesn't police.

**My count in this issue ("only 3 use a bare name filter") was right, but I nearly published it wrong.** The first version of the scanner flagged six steps. Three were real; three were `$(python .github/scripts/workspace_test_packages.py cargo-args ...)` expansions — `-p` lists, not filters. I had the right answer here by reading, and a mechanised re-take of the same question disagreed with me until I looked closer.

That produced the cell I'd actually defend in review: the scanner is mutation-tested against a synthetic `ci.yml` containing a guarded step, an option-only step, a `cargo build`, and one unguarded filtered step, asserting it finds *exactly* the last. Without that control I could not distinguish a true positive from a parser artifact — and a scanner that reports "all clear" because it parsed nothing is the same vacuous pass this issue is about.

And the tail of that: making the scanner ignore `$(...)` opens a hole where a generator emitting a bare word walks past it. That hole fails **quiet**, so it isn't skipped — each generator `ci.yml` uses is **executed** and its expansion asserted to be options only, with a non-empty guard so the check can't pass by finding nothing to check.

Before/after on the exact input demonstrated in this issue:

```
kernels::moe_renamed::   before: test result: ok. 0 passed  ->  rc=0
                          after: ::error::Test step ... executed 0 tests  ->  rc=1
```

And the three real steps under the wrapper: **1, 19, 30** tests executed, rc=0. Non-vacuous today, and they can't stop being so quietly.

Mutation results: unwrapping a step fails **2** cells (the specific one and the scanner, independently); breaking the arity check fails **7**; baseline and post-restore **33/33**.

— Pris
